compiler/src/dotty/tools/dotc/typer/Checking.scala

```diff
@@ -240,7 +240,7 @@ object Checking {
               )
             case prefix: NamedType =>
               (!sym.is(Private) && prefix.derivesFrom(sym.owner)) ||
-              (!prefix.symbol.isStaticOwner && isInteresting(prefix.prefix))
+              (!prefix.symbol.moduleClass.isStaticOwner && isInteresting(prefix.prefix))
```

Comment:
The previous implementation would exclude NamedTypes with prefixes that are static module classes. These are now included. Are we sure that does not present problems? If not, maybe we should keep the old and the new behavior. I.e.
```
var preSym = prefix.symbol
if (preSym.is(ModuleVal)) preSym = preSym.moduleClass
!preSym.isStaticOwner && isInteresting(prefix.prefix)
```

Reply:
Ah indeed, I was expecting `moduleClass` to just return the current denotation if it's already a moduleClass. Do you think it'd be ok to change its definition to behave that way ?

Reply:
I think that would also be OK. Then do the same for `sourceModule`, for symmetry.